rules/integrations/aws/credential_access_aws_getpassword_for_ec2_instance.toml

```diff
@@ -0,0 +1,108 @@
+- **Examine Related CloudTrail Events**: Search for other API calls made by the same user identity, especially those modifying security groups, network access controls, or instance metadata.
+- **Check for Lateral Movement**: Look for evidence that the obtained credentials have been used to access other resources or services within AWS.
+- **Investigate the Origin of the API Call**: Analyze the IP address and geographical location from which the request originated. Determine if it aligns with expected locations for legitimate administrative activity.
+
+### False Positive Analysis
+
+- **Legitimate Administrative Actions**: Ensure that the activity was not part of legitimate administrative tasks such as system maintenance or updates.
+- **Automation Scripts**: Verify if the activity was generated by automation or deployment scripts that are authorized to use `GetPasswordData` for legitimate purposes.
+
+### Response and Remediation
+
+- **Immediate Isolation**: If suspicious, isolate the affected instance to prevent any potential lateral movement or further unauthorized actions.
+- **Credential Rotation**: Rotate credentials of the affected instance or assumed role and any other potentially compromised credentials.
+- **User Account Review**: Review the permissions of the implicated user identity. Apply the principle of least privilege by adjusting permissions to prevent misuse.
+- **Enhanced Monitoring**: Increase monitoring on the user identity that triggered the rule and similar EC2 instances.
+- **Incident Response**: If malicious intent is confirmed, initiate the incident response protocol. This includes further investigation, containment of the threat, eradication of any threat actor presence, and recovery of affected systems.
+- **Preventative Measures**: Implement or enhance security measures such as multi-factor authentication and continuous audits of sensitive operations like `GetPasswordData`.
+
+### Additional Information
+
+Refer to resources like [AWS privilege escalation methods](https://cloud.hacktricks.xyz/pentesting-cloud/aws-security/aws-privilege-escalation/aws-ec2-privesc) and the MITRE ATT&CK technique [T1552.005 - Cloud Instance Metadata API](https://attack.mitre.org/techniques/T1552/005/) for more details on potential vulnerabilities and mitigation strategies.
+
+"""
+references = [
+    "https://cloud.hacktricks.xyz/pentesting-cloud/aws-security/aws-privilege-escalation/aws-ec2-privesc",
+]
+risk_score = 47
+rule_id = "8446517c-f789-11ee-8ad0-f661ea17fbce"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Data Source: AWS",
+    "Data Source: Amazon Web Services",
+    "Data Source: Amazon EC2",
+    "Use Case: Identity and Access Audit",
+    "Resources: Investigation Guide",
+    "Tactic: Credential Access",
+]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+event.dataset: aws.cloudtrail
```

Comment:
`GetPasswordData` with `aws.cloudtrail.error_code:"Client.UnauthorizedOperation"` looks more suspicious but may fit in a different rule.

Reply:
That is a good point. The tricky part with only signaling on unauthorized activity is the nature of adversaries leveraging valid accounts to conduct their TTPs. Always have to find a balance. I think in this case, with the Assumed Role we may catch an opportunist attempting to survey EC2 instances for credential access.